### PR TITLE
examples: rgw: add boto3 append & get usage api extensions

### PR DESCRIPTION
**Pull request by theanalyst**
- Put Object takes a `--append` `--append-position` params which allows for
  appendable objects. Also PutObject api response has a AppendPosition output
  which displays the next position to append to
- GetUsageStats api added which calls the get usage stats per user

### CODE REVIEW

**Comment by cbodley** [comment]
cc @zhangsw, this exposes the AppendObject api as an s3 extension

**Comment by cbodley** [comment]
this is really cool, @theanalyst. maybe it's time to start thinking about test coverage for these extensions in s3tests? cc @alimaredia

File: examples/boto3/service-2.sdk-extras.json [hunk 1]
```diff
@@ -139,6 +178,26 @@
             },
             "documentation":"<p>A container for object tags filtering rules.</p>"
         }
+        "GetUsageStatsOutput": {
```
Comment (yuvalif) [inline examples/boto3/service-2.sdk-extras.json]: missing comma in line 180

**Comment by yuvalif** [comment]
@theanalyst could you please add python examples?

File: examples/boto3/service-2.sdk-extras.json [hunk 2]
```diff
@@ -12,6 +12,17 @@
...
+        },
+        "GetUsageStats":{
+            "name":"GetUsageStats",
+            "http":{
+                "method":"GET",
+                "requestUri":"/?usage",
+                "responseCode":200
+            },
+            "output": {"shape": "GetUsageStatsOutput"},
+            "documentationUrl":"https://docs.ceph.com/docs/master/radosgw/s3/",
```
Comment (yuvalif) [inline examples/boto3/service-2.sdk-extras.json]: please point to the actual documentation of the feature?
is this a standard S3 feature?

File: examples/boto3/service-2.sdk-extras.json [hunk 3]
```diff
@@ -26,6 +37,34 @@
             }
         },
         "AllowUnordered":{"type":"boolean"},
+        "PutObjectRequest": {
```
Comment (yuvalif) [inline examples/boto3/service-2.sdk-extras.json]: could you please add doc URL: https://docs.ceph.com/docs/master/radosgw/s3/objectops/#append-object

File: examples/boto3/service-2.sdk-extras.json (hunk 1 above)
Comment (theanalyst) [reply examples/boto3/service-2.sdk-extras.json]: good catch

File: examples/boto3/service-2.sdk-extras.json (hunk 2 above)
Comment (theanalyst) [reply examples/boto3/service-2.sdk-extras.json]: nope, this is an RGW only extension, looks like we haven't documented it at https://docs.ceph.com/docs/master/radosgw/s3/serviceops/

Comment (yuvalif) [reply examples/boto3/service-2.sdk-extras.json]: > nope, this is an RGW only extension, looks like we haven't documented it at https://docs.ceph.com/docs/master/radosgw/s3/serviceops/

so i guess we should... can you add this as part of this PR?

Comment (theanalyst) [reply examples/boto3/service-2.sdk-extras.json]: will do

**Comment by theanalyst** [comment]
> @theanalyst could you please add python examples?

added a couple

**Comment by theanalyst** [comment]
*update- dropped a copy-paste comment in the python examples*

**Pull request merged**